JavaFXUserInterface/src/main/java/Controllers/AssetTypeInfoController.java

```diff
@@ -161,6 +203,132 @@ public void attachEvents() {
+                @Override
+                public void changed(ObservableValue<? extends Number> observableValue, Number number, Number t1) {
+                    trainValue.setText(Integer.toString((int)trainSlider.getValue()));
+                    testSlider.setMax(nbOfAssets - (int)trainSlider.getValue());
+                    trainSize = (int)trainSlider.getValue();
+                }
+            });
+            testSlider.valueProperty().addListener(new ChangeListener<Number>() {
+                @Override
+                public void changed(ObservableValue<? extends Number> observableValue, Number number, Number t1) {
+                    testValue.setText(Integer.toString((int)testSlider.getValue()));
+                    trainSlider.setMax(nbOfAssets - (int)testSlider.getValue());
+                    testSize = (int)testSlider.getValue();
+                }
+            });
+
+            try{
+                modelEvaluateBtn.setOnMouseClicked(mouseEvent -> {
+                    try {
+                        int from = (int) trainSlider.getValue()+1;
+                        int to = (int) trainSlider.getValue()+1+(int) testSlider.getValue();
+                        trainDataset  = DataPrePreprocessorController.getInstance().addRULCol(modelController.createInstancesFromAssets(assetDAO.getAssetsFromAssetTypeID(Integer.parseInt(assetType.getId())).subList(0, (int) trainSlider.getValue()-1)));
+                        testDataset = DataPrePreprocessorController.getInstance().addRULCol(modelController.createInstancesFromAssets(assetDAO.getAssetsFromAssetTypeID(Integer.parseInt(assetType.getId())).subList(from, to-1)));
+                    } catch (Exception exception) {
+                        exception.printStackTrace();
+                    }
+                    try {
+                        evaluateModels(mouseEvent);
+                    } catch (Exception exception) {
+                        exception.printStackTrace();
+                    }
+                });
+            }
+            catch (Exception e){}
+        }
+    }
+    /**
+     * Evaluates all models of a specific Asset Type
+     *
+     * @param mouseEvent is an event trigger to evlauate models
+     * @author Talal
+     */
+    private void evaluateModels(MouseEvent mouseEvent)  {
+        try {
+            ArrayList<String> models = modelDAO.getListOfModels();
+            trainDataset.setClassIndex(trainDataset.numAttributes() - 1);
+            testDataset.setClassIndex(testDataset.numAttributes() - 1);
+
+
+            for(int i=0;i<models.size();i++){
+                switch (models.get(i)) {
+                    case "Linear":
+                        LinearRegressionModelImpl linearRegressionModelImpl = new LinearRegressionModelImpl();
+                        Classifier model =  linearRegressionModelImpl.trainModel(trainDataset);
+                        calculateEvaluation(model,trainDataset,testDataset,1);
+
```

Comment:
It would be better to use the classes we already have to return the model object rather than creating new ones. It might be fine for now but when we add parameters in sprint 8, creating them this way will only create the default classifier and will not include the parameters added in the actual model classes.

Reply:
I am using the classes that we in the models package